src/BlockPatterns.php

```diff
@@ -39,15 +39,35 @@ class BlockPatterns {
 	 */
 	private $patterns_path;
 
+	/**
+	 * Info whether the theme support was already invoked.
+	 *
+	 * @var bool $theme_support_added
+	 */
+	private $theme_support_added;
+
 	/**
 	 * Constructor for class
 	 *
 	 * @param Package $package An instance of Package.
 	 */
 	public function __construct( Package $package ) {
-		$this->patterns_path = $package->get_path( 'patterns' );
+		$this->patterns_path       = $package->get_path( 'patterns' );
+		$this->theme_support_added = false;
 
+		add_action( 'after_setup_theme', array( $this, 'set_woocommerce_blocks_patterns_support' ), 5 );
```

Comment:
We need to let theme developers remove the support, by calling `remove_theme_support` as a callback to the same action. Discussed in [this comment](https://github.com/woocommerce/woocommerce-blocks/pull/9807/files#r1231102735) with Darren.

Reply:
Depends on when in the code execution their hook is registered - the first one would get executed first. This is why setting this call to `add_theme_support` is important to do on an earlier priority than the default priority. That way theme's (or other code) explicitly removing theme support on the default hook will be guaranteed to run as expected. `remove_theme_support` has to execute _after_ `add_theme_support`.

Reply:
I must of never encounter a situation where if both priorities are set to default and it didn't work (unless people are maybe putting that hook within another hook which they may be doing it wrong). I understand setting it earlier is even "safer" in terms of it being earlier than the default (10) however, people can easily set their hooks to (4) and this would also fail.

But I am not partial to making this change. Just want to understand the reasoning.